raft/raft.go

```diff
@@ -611,6 +611,7 @@ func (r *raft) becomePreCandidate() {
 	// but doesn't change anything else. In particular it does not increase
 	// r.Term or change r.Vote.
 	r.step = stepCandidate
+	r.votes = make(map[uint64]bool)
```

Comment:
vote is what we voted for, it's in hard state.  while votes is what other nodes grant/reject us, it's a memory state.
we can't change vote, but can change votes when a pre-vote happen.